include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp

```diff
@@ -204,39 +235,28 @@ class aligned_sequence_builder
 
         for (auto const & [dir, span] : rev_traces)
         {
-            switch (dir)
-            {
-                case trace_directions::diagonal:
-                {
-                    std::ranges::advance(fst_it, span);
-                    std::ranges::advance(sec_it, span);
-                    break;
-                }
-                case trace_directions::up:
-                {
-                    fst_it = insert_gap(fst_aligned, fst_it, span);
-                    std::ranges::advance(sec_it, span);
-                    break;
-                }
-                case trace_directions::left:
-                {
-                    sec_it = insert_gap(sec_aligned, sec_it, span);
-                    std::ranges::advance(fst_it, span);
-                    break;
-                }
-                case trace_directions::left_open:
-                case trace_directions::up_open:
-                case trace_directions::none:
-                default:
-                {
-                    assert(false); // This should never happen.
-                }
+            if (dir == trace_directions::up)
+                fst_it = insert_gap(fst_aligned, fst_it, span);
+
+            if (dir == trace_directions::left)
+                sec_it = insert_gap(sec_aligned, sec_it, span);
+
+            if constexpr (iterator_is_advanceable<decltype(fst_it)> && iterator_is_advanceable<decltype(sec_it)>)
```

Comment:
Why aren't you just querying the `random_access_iterator`? We should not implement an iterator that is advancable but nothing else? Then we avoid an extra concept

Reply:
But that’s the whole point of the mess with the gap decorator iterator. It is not a random access but I can use the += on it anyhow. The concept explicitly checks the iterator category.
 so I would not get the desired behavior.

Reply:
ah sorry. Can't you check random_access or psuedo_random_access?

Reply:
you mean random_access_range? That wouldn't work, since it refines the random_access_iterator concept. The pseudo_random_access would not be helpful either, since I don't use the begin_ra interface but the normal iterator interface and the psudo_random_access_range doesn't guarantee anything about the standard iterator. But in the latest #1267 there were some problems detected that I missed. I will need to refactor the class a little bit to use a policy based design. I think it also to improve it's usage. I will make this before this PR and rebase it then onto this, so this one is blocked for the time being.

Reply:
ok